drivers/dma/dma_mchp_xec.c

```diff
@@ -335,13 +335,21 @@ static int check_blocks(struct dma_xec_channel *chdata, struct dma_block_config
 static int dma_xec_configure(const struct device *dev, uint32_t channel,
 			     struct dma_config *config)
 {
-	const struct dma_xec_config * const devcfg = dev->config;
-	struct dma_xec_regs * const regs = devcfg->regs;
-	struct dma_xec_data * const data = dev->data;
+	struct dma_xec_config * const devcfg;
+	struct dma_xec_regs * const regs;
+	struct dma_xec_data * const data;
 	uint32_t ctrl, mstart, mend, dstart, unit_size;
 	int ret;
 
-	if (!dev || !config || (channel >= (uint32_t)devcfg->dma_channels)) {
+	if (!dev || !config) {
```

Comment:
I think the `dev` would already have to point to a valid device, otherwise the `api` member can't be accessed in `dma_config`.

Reply:
@pdgendt: i'm not following - i can't see any `api` access in this file. are you meaning code somewhere else?

Reply:
This function is called from `dma_config` in `include/zephyr/drivers/dma.h` which already dereferences `dev`:

```c
static inline int dma_config(const struct device *dev, uint32_t channel,
			     struct dma_config *config)
{
	const struct dma_driver_api *api =
		(const struct dma_driver_api *)dev->api;

	return api->config(dev, channel, config);
}
```

If `dev == NULL` would be true, the API call behavior would be undefined.

Reply:
right,
so the correct change is probably to just remove the (redundant) null check on `dev`?

Reply:
IMO, yes, which also doesn't require to change the `devcfg/regs/data` assignments.